### ISSUE TITLE
K: drive not found by Windows 98 SE

### ISSUE THREAD

**Issue by skipcress** [joncampbell123/dosbox-x#3608]
### Question

I am trying to mount a K: CD-ROM drive, and it's mounting fine in DOS, but Windows 98 SE isn't enumerating it. I've got "[ide, primary]" on through "[ide, octernary]" enabled (my full .conf file is attached).

Any idea why I can't mount a K: drive in Windows?

Thanks!
Kirk

[win98se.conf.txt](https://github.com/joncampbell123/dosbox-x/files/9033827/win98se.conf.txt)

### Have you checked that no similar question(s) exist?

- [X] I have searched and didn't find any similar question.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by Torinde** [comment]
K is your 9th image, so I assume your [ide, quinternary] isn't setup properly. See #3557.

Each IDE controller supports two drives - Master and Slave (e.g. with two IDE controllers you get: IDE Primary Master, IDE Primary Slave, IDE Secondary Master, IDE Secondary Slave). Primary and Secondary are default (occupying IRQ 14 and 15), so up to 4th image (1 HDD + 3 CDs) normally work without problems.

Your conf file seems too bare - there is no irq/io/altio in the IDE controller sections. Per [default IRQs list](https://en.wikipedia.org/wiki/Interrupt_request_(PC_architecture)) on physical PCs I would assume Tertiary and Quaternary would be easy to add (IRQ 10 and 11). And that's what you observe - eight images loading good.

For the Quinternary probably you have to assign IRQ, etc. manually and make sure it isn't utilized by something else.

**Comment by maron2000** [comment]
Try adding -ide option when mounting your image with IMGMOUNT.
Refer to this issue #3418 for details.

**Comment by skipcress** [comment]
I was able to circumvent the issue by modding one of the games configurations such that it no longer required a CD mounted, and if I do in fact require more CD drives I'll try the resolutions recommended above - thanks!

**Comment by skipcress** [comment]
@Torinde had it right. In order to add another drive:

1. I went into "Settings" -> "Control Panel" -> "Add New Hardware" and added a new Hard Disk Controller of type "Standard IDE/ESDI Hard Disk Controller," taking note of the IRQ and Input/Output range assigned to it
2. In my config file, I set the irq, io, and altio for the quinternary ide controller (io was set to the first number of the "Input/Output Range," and altio was set to the second number in the "Input/Output Range")

After that, the drive was detected.
